### PR TITLE
feature flag read recents from read replica

### PR DESCRIPTION
**Pull request by camallen**
some of the recents queries can be costly as they invert the order on created at before selecting the first page. push this to the read replica where we can

# Review checklist

- [ ] First, the most important one: is this PR small enough that you can actually review it? Feel free to just reject a branch if the changes are hard to review due to the length of the diff.
- [ ] If there are any migrations, will they the previous version of the app work correctly after they've been run (e.g. the don't remove columns still known about by ActiveRecord).
- [ ] If anything changed with regards to the public API, are those changes also documented in the `apiary.apib` file?
- [ ] Are all the changes covered by tests? Think about any possible edge cases that might be left untested.

### CODE REVIEW

File: app/controllers/concerns/recents.rb [hunk 1]
```diff
@@ -2,10 +2,24 @@ module Recents
...
+        ps,
+        Recent.where(:"#{resource_name}_id" => resource_ids),
+        { url_prefix: "#{resource_sym.to_s.pluralize}/#{resource_ids}" }
+      )
+    end
+  end
+
+  private
+
+  def read_recents_from_database
```
Comment (marten) [inline app/controllers/concerns/recents.rb]: Wonder if we should just turn this into a globally available `Panoptes.read_from_replica(flipper_key)` method? 2/10 tho

Comment (camallen) [reply app/controllers/concerns/recents.rb]: thought about it as well, do we want granularity to have the dumps run on the replica but switch this feature back (say the replica lag is really high)?

Comment (marten) [reply app/controllers/concerns/recents.rb]: Maybe? That's why I was passing in the flipper key, but I'm happy to leave it off if you think we don't need it.

Comment (camallen) [reply app/controllers/concerns/recents.rb]: ha! bad cam, I read that completely differently. i'll extract the common code block to something like that.

File: app/workers/concerns/dump_worker.rb [hunk 2]
```diff
@@ -128,13 +128,7 @@ def get_resource
...
-  def read_from_database
-    if Panoptes.flipper.enabled?("dump_data_from_read_slave")
-      Slavery.on_slave do
-        yield
-      end
-    else
-      yield
-    end
+  def read_from_database(&block)
+    DatabaseReplica.read("dump_data_from_read_slave", &block)
```
Comment (camallen) [inline app/workers/concerns/dump_worker.rb]: this is costlier than using a block as it converts the block to a proc but it keeps the code cleaner.

**Comment by marten** [comment]
LGTM

**Pull request merged**